### PR TITLE
ENH: add SeqsData class method for construction from dict

### PR DESCRIPTION
**Pull request by KatherineCaley**
## Summary by Sourcery

Introduce a 'from_seqs' class method for SeqsData and AlignedSeqsData to standardize construction from a dictionary and alphabet, replacing the previous 'from_aligned_seqs' method. Update tests to validate the new method and ensure correct behavior with different alphabets.

Enhancements:
- Introduce a class method 'from_seqs' for SeqsData and AlignedSeqsData to facilitate construction from a dictionary and alphabet.

Tests:
- Add tests to verify the construction of SeqsData from a dictionary and alphabet, including cases with incorrect alphabets.

### CODE REVIEW

File: src/cogent3/core/new_alignment.py [hunk 1]
```diff
@@ -276,19 +275,33 @@ def __init__(
...
         for name, seq in data.items():
             arr = self._alphabet.to_indices(seq)
             arr.flags.writeable = False
             self._data[str(name)] = arr
-        self._offset = offset or {}
-        if check:
-            assert set(self._offset.keys()) <= set(
-                self._data.keys()
-            ), "sequence name provided in offset not found in data"
         self._reversed = reversed
```
[the remark this answers is not in the document]
Comment (KatherineCaley) [reply src/cogent3/core/new_alignment.py:292]: already marked with a refactor target, will address in future PR!

File: tests/test_core/test_new_alignment.py [hunk 2]
```diff
@@ -157,6 +163,21 @@ def make_typed(seq, data_type, moltype):
...
+    sd = new_alignment.SeqsData(data=str_seqs_dict, alphabet=dna_alphabet)
+    assert isinstance(sd, new_alignment.SeqsData)
+
+    sd = new_alignment.SeqsData.from_seqs(data=str_seqs_dict, alphabet=dna_alphabet)
+    assert isinstance(sd, new_alignment.SeqsData)
+
+
+def test_seqs_data_construction_wrong_alphabet(str_seqs_dict, rna_alphabet):
+    """SeqsData should raise ValueError if alphabet is incompatible with data"""
+    with pytest.raises(ValueError):
```
Comment (GavinHuttley) [inline tests/test_core/test_new_alignment.py:177]: we should raise `new_alphabet.AlphabetError` instead

File: tests/test_core/test_new_alignment.py [hunk 3]
```diff
@@ -2238,9 +2258,8 @@ def test_sequence_collection_distance_matrix_raises_wrong_moltype(moltype):
 
 @pytest.mark.parametrize("moltype", ("dna", "rna"))
 def test_sequence_collection_distance_matrix_passes_correct_moltype(moltype):
-    data = {"s1": "ACGTA", "s2": "ACGTA"}
+    data = {"s1": "ACGA", "s2": "ACGA"}
     seqs = new_alignment.make_unaligned_seqs(data, moltype=moltype)
-
     seqs.distance_matrix()
```
Comment (GavinHuttley) [inline tests/test_core/test_new_alignment.py:2263]: no assertion in this test?

**Review by GavinHuttley — CHANGES_REQUESTED** [review]
Some tweaks. I also think we should be raising AlphabetError when there's a sequence is inconsistent with the alphabet.

**Review by GavinHuttley — APPROVED** [review]
Awesome 🚀

**Pull request merged**